oximeter/db/src/client.rs

```diff
@@ -2282,157 +2778,141 @@ mod tests {
         let datum = Datum::I16(1);
         let as_json = serde_json::Value::from(1_i16);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_u16_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_measurement_u16_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::U16(1);
         let as_json = serde_json::Value::from(1_u16);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_i32_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_measurement_i32_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::I32(1);
         let as_json = serde_json::Value::from(1_i32);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_u32_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_measurement_u32_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::U32(1);
         let as_json = serde_json::Value::from(1_u32);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_i64_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_measurement_i64_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::I64(1);
         let as_json = serde_json::Value::from(1_i64);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_u64_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    async fn recall_measurement_u64_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::U64(1);
         let as_json = serde_json::Value::from(1_u64);
-        test_recall_measurement_impl::<u8>(address, datum, None, as_json)
+        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
             .await?;
         Ok(())
     }
 
-    async fn recall_measurement_f32_test(
-        address: SocketAddr,
-    ) -> Result<(), Error> {
+    // NOTE: This test is ignored intentionally.
+    //
+    // For some reason when the value is an f64 it inserts in both measurements_f32
+    // and measurements_f64 tables. This behaviour is unexpected, and should be investigated
```

Comment:
This is not something we can ignore in this PR. We can't be inserting measurements into the wrong table. I've looked through the code, and am not sure how this could happen. Do we see it with any other measurement types? The test is run and passes on `main`, so might it be possible that this was due to multiple tests run concurrently?

Reply:
Could you please point me to the commit / line which fixes this one?

Reply:
https://github.com/oxidecomputer/omicron/pull/4372/commits/0b77159cfe6184417be62ac01b13e6e7537087c2#diff-d7e0c19781820dffa1f107bb7e5b9b78bc9f79d85255c3457010f989987d83d3R209